src/surveys/components/Survey.tsx

```diff
@@ -44,80 +52,102 @@ export default function Survey({
-      const index = survey.questions.findIndex(question => question.id === activeQuestion.id);
-      console.log('resolveNextQuestion', activeQuestion.id, index);
+      if (survey.questions.length === 0) {
+        return null;
+      }
 
-      if (index >= 0 && ((index + 1) < survey.questions.length)) {
-        nextQuestion = survey.questions[index + 1];
+      const nextQuestionId = getNextQuestionId(activeQuestion, answers);
+
+      let nextQuestion = null;
+      if (nextQuestionId) {
+        nextQuestion = survey.questions.find(
+          (question) => question.id === nextQuestionId
+        );
+        console.log('resolveNextQuestion', nextQuestionId, nextQuestion);
+      } else {
+        const index = survey.questions.findIndex(
+          (question) => question.id === activeQuestion.id
+        );
+        console.log('resolveNextQuestion', activeQuestion.id, index);
+
+        if (index >= 0 && index + 1 < survey.questions.length) {
+          nextQuestion = survey.questions[index + 1];
+        }
       }
-    }
 
-    return nextQuestion;
-  }, [survey, activeQuestion])
+      return nextQuestion;
+    },
+    [survey, activeQuestion]
+  );
 
-  const isLastQuestion = useCallback((questionId: ID) => {
-    return questionId === survey.questions[survey.questions.length - 1].id;
-  }, [survey])
+  const isLastQuestion = useCallback(
+    (questionId: ID) => {
+      return questionId === survey.questions[survey.questions.length - 1].id;
+    },
+    [survey]
+  );
 
-  const onAnswered = useCallback((answers: SurveyAnswer[]) => {
-    if (!survey) {
-      return;
-    }
+  const onAnswered = useCallback(
+    (answers: SurveyAnswer[]) => {
+      if (!survey) {
+        return;
+      }
 
-    setLoading(true);
+      setLoading(true);
 
-    const nextQuestion = resolveNextQuestion(answers);
-    const finished = isLastQuestion(activeQuestion.id);
+      const nextQuestion = resolveNextQuestion(answers);
+      const finished = isLastQuestion(activeQuestion.id);
 
-    if (answers.length > 0) {
-      answers[answers.length - 1].finished = finished;
-    }
+      if (answers.length > 0) {
+        answers[answers.length - 1].finished = finished;
+      }
 
-    onQuestionAnswered?.(survey.id, activeQuestion.id, answers);
+      onQuestionAnswered?.(survey.id, activeQuestion.id, answers);
 
-    setLoading(false);
+      setLoading(false);
 
-    if (finished || !nextQuestion) {
-      onSurveyCompleted?.(survey.id);
+      if (finished || !nextQuestion) {
+        onSurveyCompleted?.(survey.id);
 
-      close();
-      return;
-    }
+        close();
+        return;
+      }
 
-    setActiveQuestion(nextQuestion);
-  }, [survey, activeQuestion]);
+      setActiveQuestion(nextQuestion);
+    },
+    [survey, activeQuestion]
+  );
 
   return (
-    <div>
-      {survey.settings.showProgressBar && <Progress progress={progress} bgColor={survey.theme?.progressBar ?? '#050505'} />}
+    <Wrapper
+      close={forceClose}
```

Comment:
@AfeezGL Instead of `close={forceClose}`, you could do `close={() => close(true)}`.